File: frontend/src/constants/services.ts
import { ServiceItem } from "../types";

export const SERVICE_CATALOG: ServiceItem[] = [
  { id: "00000000-0000-0000-0000-000000000001", name: "Thuê sân", unit_price: 30000 },
  { id: "00000000-0000-0000-0000-000000000002", name: "Nước suối", unit_price: 10000 },
  { id: "00000000-0000-0000-0000-000000000003", name: "Khăn lạnh", unit_price: 5000 },
];

export interface ServiceCreate {
  name: string;
  unit_price: number;
  quantity: number;
}

File: app/schemas/services.py
from pydantic import BaseModel
from typing import Optional
from uuid import UUID
from fastapi import HTTPException
from app.schemas.category import CategoryResponse  # 🧩 Import category schema

# ✅ Schema khi tạo/ghi service (POST) - hỗ trợ upsert với id optional
class ServiceCreate(BaseModel):
    id: Optional[UUID] = None  # 🆕 Thêm id để backend nhận diện bản ghi update/create
    name: str
    unit_price: int
    quantity: Optional[int] = 0  # ✅ Cho phép null
    category_id: Optional[UUID] = None  # UUID đồng bộ với DB

# ✅ Schema cho update service (PUT) - optional để partial update
class ServiceUpdate(BaseModel):
    name: Optional[str] = None
    unit_price: Optional[int] = None
    quantity: Optional[int] = 0  # ✅ Cho phép null
    category_id: Optional[UUID] = None

# ✅ Schema trả về (GET) - kèm category nếu có
class ServiceItem(BaseModel):
    id: UUID
    name: str
    unit_price: int
    quantity: Optional[int] = 0  # ✅ Cho phép null
    category: Optional[CategoryResponse] = None

    class Config:
        orm_mode = True
